lib/phoenix/digester.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Phoenix.Digester do
+  @digested_file_regex ~r/(-[a-fA-F\d]{32})/
+
+  @moduledoc """
+    Digests and compress assets files.
+
+    For each file under the given input path, Phoenix will generate a digest
+    and also compress in `.gz` format. The filename and its digest will be
+    used to generate the manifest file. It also avoid duplications checking
+    for already digested files.
+  """
+
+  @doc """
+  Compile the assets and save them in the given output path.
+
+    * `input_path` - The path where the assets are located
+    * `output_path` - The path where the compiled/compressed files will be saved
+  """
+  @spec compile(String.t, String.t) :: :ok | {:error, :invalid_path}
+  def compile(input_path, output_path) do
+    if File.exists?(input_path) && File.exists?(output_path) do
```

Comment:
The output path does not need to exist. Just call `File.mkdir_p!(output_path)` to ensure we have created it.